win32/src/threadpool_win32.c

```diff
@@ -87,7 +87,19 @@ static VOID CALLBACK on_work_callback(PTP_CALLBACK_INSTANCE instance, PVOID cont
 
 static VOID CALLBACK on_work_callback_v2(PTP_CALLBACK_INSTANCE instance, PVOID context, PTP_WORK work)
 {
-    LogWarning("Printing %p %p %p", instance, context, work);
+    if (context == NULL)
+    {
+        /* Codes_SRS_THREADPOOL_WIN32_05_001: [ If context is NULL, on_work_callback_v2 shall return. ]*/
+        LogError("Invalid arguments: PTP_CALLBACK_INSTANCE instance=%p, PVOID context=%p, PTP_WORK work=%p",

```

Comment:
>LogError [](http://example.com/codeflow?start=8&length=8)

This should be a fatal error, it should be LogCriticalAndTerminate

Reply:
Hi Jelani, it's a callback function, do we want to exit Fatal for callbacks? Also, reference `on_work_callback` where we are just doing `LogError`

Reply:
We would terminate where because they context should never be NULL and if it is then something really bad has happened and we can't recover from this.